WPFHospitalEditor/Building.xaml.cs

```diff
@@ -158,8 +206,27 @@ private void selectMapObject(object sender, MouseButtonEventArgs e)
         
         private void openAdditionalInformationDialog(MapObject mapObject)
         {
-            AdditionalInformation additionalInformation = new AdditionalInformation(mapObject);
-            additionalInformation.ShowDialog();
+            int index = floor.SelectedIndex;
+            AdditionalInformation additionalInformation = new AdditionalInformation(mapObject, this, index);
+            additionalInformation.Owner = this;
+            additionalInformation.ShowDialog(); 
+        }
+
+        public Floor getFloor(int index)
+        {
+            return buildingFloors[index];
+        }
+
+        public List<MapObject> getAllBuildingObjects()
+        {
+            return allBuildingObjects;
+        }
+
+        public static String getBuildingId(MapObject mapObject)
+        {
+            String[] firstSplit = mapObject.Description.Split("&");
+            String[] secondSplit = firstSplit[0].Split("-");
+            return secondSplit[0];
```

Comment:
get method for building id shouldn't be static

Reply:
It is fixed.